packages/node_modules/@webex/webex-core/src/lib/services/services.js

```diff
@@ -327,20 +327,49 @@ const Services = WebexPlugin.extend({
    * @returns {object}
    */
   _formatReceivedHostmap(serviceHostmap) {
-    const output = [];
-    let item = {};
+    let serviceItem;
 
-    Object.keys(serviceHostmap.serviceLinks).forEach((key) => {
-      item = {};
+    // map home cluster service hosts
+    const mappedLinks = Object.keys(serviceHostmap.serviceLinks).map((key) => {
+      serviceItem = {};
 
-      item.name = key;
-      item.defaultUrl = serviceHostmap.serviceLinks[key];
-      item.hosts = serviceHostmap.hostCatalog[Url.parse(item.defaultUrl).host];
+      serviceItem.name = key;
+      serviceItem.defaultUrl = serviceHostmap.serviceLinks[key];
+      serviceItem.defaultHost = Url.parse(serviceItem.defaultUrl).hostname;
+      serviceItem.hosts = serviceHostmap.hostCatalog[serviceItem.defaultHost];
 
-      output.push(item);
+      serviceItem.hosts = serviceItem.hosts.map((host) => ({
+        ...host,
+        homeCluster: true
+      }));
+
+      return serviceItem;
+    });
+
+    // collect unmapped cluster information
+    const unmappedHosts = Object.keys(serviceHostmap.hostCatalog).filter(
+      (key) => mappedLinks.every(
+        (mappedLink) => mappedLink.defaultHost !== key
+      )
+    )
+      // map the filtered hosts against their service names
+      .map((key) => ({
+        serviceName: serviceHostmap.hostCatalog[key][0].id.split(':')[3],
+        hosts: serviceHostmap.hostCatalog[key]
+      }));
```

Comment:
Instead of looping over twice with `filter` and `map`, you can do it once using `reduce`